chrome/content/zotero/zoteroPane.js

```diff
@@ -4794,6 +4869,102 @@ var ZoteroPane = new function()
 			}
 		}
 	};
+
+
+	/**
+	 * Attempt to find a file in the LABD matching the passed attachment
+	 * by searching successive subdirectories. Prompt the user if a match is
+	 * found and offer to relink one or all matching files in the directory.
+	 * The user can also choose to relink manually, which opens a file picker.
+	 *
+	 * If the path is 'C:\Users\user\Documents\Dissertation\Files\Paper.pdf'
+	 * and the LABD is '/Users/user/Documents', check:
+	 *
+	 * 1. /Users/user/Documents/Paper.pdf
+	 * 2. /Users/user/Documents/Dissertation/Paper.pdf
+	 * 3. /Users/user/Documents/Dissertation/Files/Paper.pdf
```

Comment:
I specified it this way, but I wonder if we have this backward. If the LABD is ~/Documents, shouldn't it try starting with the full path?

1. /Users/user/Documents/Users/user/Documents/Dissertation/Files/Paper.pdf
2. /Users/user/Documents/user/Documents/Dissertation/Files/Paper.pdf
3. /Users/user/Documents/Documents/Dissertation/Files/Paper.pdf
4. **/Users/user/Documents/Dissertation/Files/Paper.pdf**

Or, even better, we could split normalized paths into components and remove matching components, in which case we'd get the above example in the first try:

1. Path is `['Users', 'user', 'Documents', 'Dissertation', 'Files', 'Paper.pdf']`
2. Remove LABD of `['Users', 'user', 'Documents']`
3. End up with `['Dissertation', 'Files', 'Paper.pdf']`, which is the correct relative path

Otherwise we're risking matching the wrong file if there's another Paper.pdf in ~/Documents. (Obviously linked files will generally be named more clearly, but still.) We'll need to make sure we're catching errors from paths that are too long to try (which we're probably doing already, but I can't remember).

Reply:
Yeah, looking this over last night I realized that it was a contrived example and splitting would almost never help. So just the starting with the full path makes sense.